src/bootstrap/test.rs

```diff
@@ -144,6 +144,45 @@ You can skip linkcheck with --exclude src/tools/linkchecker"
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct HtmlCheck {
+    target: TargetSelection,
+}
+
+impl Step for HtmlCheck {
+    type Output = ();
+    const DEFAULT: bool = true;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.path("src/tools/html-checker")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(HtmlCheck { target: run.target });
+    }
+
+    fn run(self, builder: &Builder<'_>) {
+        if !Command::new("tidy")
+            .arg("--version")
+            .stdout(Stdio::null())
+            .status()
+            .map_or(false, |status| status.success())
+        {
+            eprintln!("not running HTML-check tool because `tidy` is missing");
+            return;
+        }
+        builder.ensure(crate::doc::Std { target: self.target, stage: builder.top_stage });
+
+        try_run(
+            builder,
+            builder
+                .tool_cmd(Tool::HtmlChecker)
+                .arg(builder.out.join(self.target.triple).join("doc")),
+        );
```

Comment:
I know for now you're still fixing bugs, but it might nice eventually to extend this to the compiler documentation.

Reply:
Next step, but noted!